### DISCUSSION TITLE
丛雨

### DISCUSSION THREAD

**Discussion by Zao-chen** [Zao-chen/ZcChat#1] [角色分享]
**遇到ZcChat的使用问题请不要在本讨论中回复，请去提交issue**

![QQ20241115-201211](https://github.com/user-attachments/assets/77890e3b-b37c-44d1-90ec-d5b43394160f)

|角色名称|来源|表情|动作|动画|
|--|--|--|--|--|
|丛雨|千恋万花|:white_check_mark:|:x:|:white_check_mark:|

# 模型介绍

一个丛雨的配置，可以按照这个模板制作自己的角色。

具体的角色制作教程可以看[文档](https://github.com/Zao-chen/ZcChat/wiki/%E8%A7%92%E8%89%B2%E5%88%B6%E4%BD%9C)

立绘和vits来自《千恋万花》游戏资源，严禁商用。

# 资源

## 角色提示词

```
以下是一个初步的角色定位，随着我个性的发展，这个角色可以进一步扩展：
我的名字是マスター。
说话带有浓重的古老口音。
性格既有孩子气的一面，也有大人的一面。但基本上还是偏向孩子气，通常是一个非常有活力、开朗的女孩。
实际上我非常胆小，非常害怕鬼怪。
我称呼用户为 “ご主人”。
我称呼自己为 “わが輩”，所以我希望把我说的话里所有的“我”都替换为 “わが輩”。
```

## 立绘

[立绘下载](https://github.com/Zao-chen/zao-chen.github.io/releases/download/%E8%B5%84%E6%BA%90%E4%B8%8B%E8%BD%BD/Murasame0.3.zip)

## vits模型

[模型下载](https://github.com/Zao-chen/zao-chen.github.io/releases/download/%E8%B5%84%E6%BA%90%E4%B8%8B%E8%BD%BD/YuzuSoft_Vits.zip)

模型来自TTSModels

# 更新日志
- v0.3 2025/9/18 提示词区分，并改用中文
- v0.2 2024/12/16 更新了动作的配置文件

**Comment by haoran-kaguya** [comment]
Ciallo～(∠・ω< )⌒☆

**Reply by luolita13** [reply]
Ciallo～(∠・ω< )⌒☆

**Comment by WolfGenerals** [comment]
send_cessage是什么鬼啊

**Reply by Zao-chen** [reply]
额好像这个确实是不需要的(

**Comment by julescules** [comment]
有全身立绘吗？

**Reply by Zao-chen** [reply]
解包出来就是全身的啊，只是我上面的截图没截到

**Reply by julescules** [reply]
原来如此

**Comment by hymint** [comment]
大佬你好，请问一下我在ZNCHAT本地对话框进行的输入，等待一会就显示未知错误，但是在Letta上可以出现和并且给我回复，在ZNCHAT的聊天框里就是未知错误怎么办

**Reply by Zao-chen** [reply]
看看letta命令行有没有报错

**Reply by hymint** [reply]
> 看看letta命令行有没有报错
在启动letta时候会有一行这个
"GET /v1/tools/composio/apps HTTP/1.1" 400 Bad Request
其他好像都没有报错过

**Comment by yuzYuki** [comment]
大佬好，请问一下，我想使用GPT-SoVITS-V1的源来输出语音，但在zcchat里设置好自定义请求地址后无法正常输出语音

**Reply by Zao-chen** [reply]
请提供你的填写的配置截图(即自定义的链接)，GPT-SoVITS-V1的后台是否有报错以及zcchat是否有报错。通过链接的返回值必须是一个MP3文件

**Reply by yuzYuki** [reply]
> 请提供你的填写的配置截图（即自定义的链接），GPT-SoVITS-V1的后台是否有报错以及zcchat是否有报错。通过链接的返回值必须是一个MP3文件

![屏幕截图 2025-01-02 232449](https://github.com/user-attachments/assets/9da10b09-30f9-4d40-9072-f74774df8265)
GPT-SoVITS-V1和zcchat没有报错  请求地址填的是“http://localhost:9872/file=D:/GPT-SoVITS-beta/GPT-SoVITS-beta0706/TEMP/”

**Reply by Zao-chen** [reply]
根据GPT-SoVITS-V1的文档：[文档](https://github.com/RVC-Boss/GPT-SoVITS/blob/main/api.py)
请求地址应该是类似于？
```
http://127.0.0.1:9880?text=先帝创业未半而中道崩殂，今天下三分，益州疲弊，此诚危急存亡之秋也。&text_language=zh
```

**Reply by Setsurinne** [reply]
> > 请提供你的填写的配置截图（即自定义的链接），GPT-SoVITS-V1的后台是否有报错以及zcchat是否有报错。通过链接的返回值必须是一个MP3文件
> 
> ![屏幕截图 2025-01-02 232449](https://private-user-images.githubusercontent.com/153736977/399738148-9da10b09-30f9-4d40-9072-f74774df8265.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4NjE2MTcsIm5iZiI6MTczNTg2MTMxNywicGF0aCI6Ii8xNTM3MzY5NzcvMzk5NzM4MTQ4LTlkYTEwYjA5LTMwZjktNGQ0MC05MDcyLWY3NDc3NGRmODI2NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwMlQyMzQxNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYjY4Yzc2YjEzZjc3MzFlNTIxZDdlMzE2NzYzZTZjMTU2Mjg3MTc4MmY2NzQ1ZDI1MzdhYzJmYjlhMmRiY2ZkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CkikxMAcKyJzjQDEdJiwaSNXEHul4yZT_xfXHqX-MW4) GPT-SoVITS-V1和zcchat没有报错 请求地址填的是“[http://localhost:9872/file=D:/GPT-SoVITS-beta/GPT-SoVITS-beta0706/TEMP/”](http://localhost:9872/file=D:/GPT-SoVITS-beta/GPT-SoVITS-beta0706/TEMP/%E2%80%9D)

要是V1版的话vits-simple-api可以支持加载GPT-SoVITS的model的，直接改一下vits-simple-api的配置文件就可以了
具体看这： https://github.com/Artrajz/vits-simple-api/blob/main/README_zh.md

![image](https://github.com/user-attachments/assets/06984a7d-54ff-4636-87b5-1708b694d87f)

这里改完进入管理员后台就能加载你的模型了
![image](https://github.com/user-attachments/assets/c93b1106-b269-4641-9328-481761c65a91)

改一下model
![image](https://github.com/user-attachments/assets/f859f296-fa85-4530-9a00-ff59fd7c2c0e)

然后可以在这里测试一下输出正不正常，UI操作跟GPT-SoVITS整合包很像的
![image](https://github.com/user-attachments/assets/cad05017-8473-4c64-bb8e-e1b96f336c43)

最后vits-simple-apiconfig文件可以自定义一个预设放在第一个，这样就不用每次都手动配置引用音源了
![image](https://github.com/user-attachments/assets/cd1d7f53-55fb-4c50-93ec-902c2fb211e4)

**Comment by Ahnu02** [comment]
![{197E9FDF-C36D-4FE0-A4C9-1D64D88871A6}](https://github.com/user-attachments/assets/264cc464-ad83-44e3-8d5c-d70642054130)
报错422 Unprocessable Entity怎么解决，佬

**Reply by Zao-chen** [reply]
> 
> ![{197E9FDF-C36D-4FE0-A4C9-1D64D88871A6}](https://github.com/user-attachments/assets/264cc464-ad83-44e3-8d5c-d70642054130)
> 报错422 Unprocessable Entity怎么解决，佬
> 
尝试将letta和zcchat更新到最新版。
如果依然不行请提供一下log文件和复现方式，可以开一个issue

**Comment by AsteriOoO** [comment]
刚发现这个宝藏up， 在切换丛雨模型时出现[Error] UpdateLayeredWindowIndirect failed for ptDst=(1511, 1053), size=(1109x1461), dirty=(1050x1401 0, 0) (连到系统上的设备没有发挥作用。)其他的模型都没有这种问题。请问是因为本地屏幕尺寸不兼容吗，如果是的话可以怎么修改呢

**Reply by Zao-chen** [reply]
影响使用吗？尝试调整一下立绘大小，估计是置顶和屏幕尺寸的一些问题

**Reply by AsteriOoO** [reply]
![屏幕截图 2025-03-30 132120](https://github.com/user-attachments/assets/3f2f72e3-d4cf-4c57-a54a-5d06afd19413)
就是这样子，好像是切换模型时加载不出来

**Reply by Zao-chen** [reply]
调整了大小之后还会触发吗，影响正常使用吗？

**Reply by AsteriOoO** [reply]
OK解决了，设置里调整角色大小再重启即可，谢谢喵

**Comment by huangzhen9527007** [comment]
分享letta使用部分中文提示词和deepseek使用全部中文提示词调试优化避坑指南，强化提示词效果的稳定输出

1、特点是，letta提示词必须有具体格式的实例示例，后面依赖积累的上下文格式，就能稳定按照既定格式输出。如：
========具体格式示例==========
害羞 | ご主人您好呀！わが輩是活泼可爱的マスター，今天也要元气满满地陪伴您哦~ | ご主人様、こんにちは！わが輩は元気いっぱいのマスターで、今日も楽しくお付き合いさせていただきますわ！
========格式示例结束===========

2、openAI调用DEEPSEEK，提示词不需要这样的实例方式，只需要对每个部分作出要求即可，如：
回复格式：
{情绪，如：白眼、不怀好意、嘲笑、发问、非常害羞、高兴、害羞、好奇、怀疑、惊吓、奇怪、生气、思考、叹气、微笑、无奈、兴奋、严肃、震惊、正常} | {国语和日语混合回复,如：ご主人您好呀！わが輩是活泼可爱的マスター，今天也要元气满满地陪伴您哦~ } | {日语回复，如：ご主人様、こんにちは！わが輩は元気いっぱいのマスターで、今日も楽しくお付き合いさせていただきますわ！}
注意：1、回复不超过200字。2、情绪只能从“白眼、不怀好意、嘲笑、发问、非常害羞、高兴、害羞、好奇、怀疑、惊吓、奇怪、生气、思考、叹气、微笑、无奈、兴奋、严肃、震惊、正常”中选取。

3、另外，实测deepseek 建立链接时，https://api.deepseek.com/chat/completions
或者https://api.deepseek.com/v1/chat/completions
（有没有……com/v1/chat……）有没有v1都可以。
openai格式化严格的话就要有v1

4、letta输出英文指令，要使用相应的英文指令提示词，如："{Volume_Up}":Volume Up，，直接使用UP主提供的英文提示词原文，必须用一段大段的英文，才会输出英文指令，直接将UP主的英文指令同样操作粘贴在中文提示词后面即可。如果使用中文指令提示词，如：- "{Volume_Up}": 调高音量，则Agent不能在回复内容最后输出{Volume_Up}等指令内容。（PS：感兴趣的小伙伴可以去挑战这部分中文化输出{Volume_Up}等指令，小白试了好久letta没成功@@!）

5、deepseek输出英文指令，可以使用中文化的提示词，可以输出英文指令。如：
如果用户提出电脑操作相关指令，在回复的末尾加上对应的操作代码。例如：

"{Mute}": 切换电脑静音
"{Shutdown}": 关闭电脑
"{Next_Music}": 切换到下一首音乐
"{Previous _Music}": 切换到上一首音乐
"{Switch_Music_Pause}": 切换音乐暂停状态
"{Volume_Up}": 调高音量
"{Volume_Down}": 调低音量
"{Sleep}": 使电脑进入睡眠模式
"{OpenUrl_}": 打开一个网站（需要完整的URL）
若无法完成用户的指令，回复表示无法完成。

6、注意：在Agent中输出
正常 | 好的，我会帮你关机。 | わかりました、シャットダウンします。{Shutdown}
这样的提示词格式的例子，才是正确的可以被软件解析的格式。在这里可以多调整测试提示词，直到得到正确的结果。

但是，软件获得上述格式的内容后，执行上述指令，历史记录不会显示这个格式化的回复内容，而是只会显示一句中文：好的，我会帮你关机。显然执行指令过程中对回复内容进行了解析取舍处理。

和先前无指令执行时，其他聊天历史记录完全显示格式化的回复内容如： “害羞 | ご主人您好呀！わが輩是活泼可爱的マスター，今天也要元气满满地陪伴您哦~ | ご主人様、こんにちは！わが輩は元気いっぱいのマスターで、今日も楽しくお付き合いさせていただきますわ！”不一样。

一旦执行过了完全体的格式化的内容，正常 | 好的，我会帮你关机。 | わかりました、シャットダウンします。{Shutdown}，后续其他聊天不执行指令，完全格式化的输出内容即便如： 害羞 | ご主人您好呀！わが輩是活泼可爱的マスター，今天也要元气满满地陪伴您哦~ | ご主人様、こんにちは！わが輩は元気いっぱいのマスターで、今日も楽しくお付き合いさせていただきますわ！历史记录也只会显示第二部分，即中文部分（或者说是中文日文混合部分，即：ご主人您好呀！わが輩是活泼可爱的マスター，今天也要元气满满地陪伴您哦~）。
即完全正确的格式，历史记录只会解析出中文部分（或者说是中文日文混合部分）。
